src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -210,7 +210,11 @@ impl<'a> JobQueue<'a> {
             // being a compiled package
             Dirty => {
                 self.printed.insert(pkg);
-                try!(config.shell().status("Compiling", pkg));
+                //check if documenting or compiling
+                match doc {
+                    &true => try!(config.shell().status("Documenting", pkg)),
```

Comment:
I think that the logic may want to be beefed up a bit here. This means that each package *either* prints whether it's being documented or compiled, but it's also possible that both are done for a package (e.g. deps are both compiled and documented by default).

Perhaps the `printed` set could be split into `documented` and `compiled`, and then a package can both be printed as being documented and being compiled.